source/js/Directions.js

```diff
@@ -0,0 +1,188 @@
+      let listItem = document.createElement('li');
+      listItem.textContent = `${directionList[i].name}`;
+      list.appendChild(listItem);
+
+      // If there are inner steps, display them as well
+      if (directionList[i].itemListElement != undefined) {
+        for (let j = 0; j < directionList[i].itemListElement.length; j++) {
+          let innerListItem = document.createElement('li');
+          innerListItem.textContent = `${directionList[i].itemListElement[j].text}`;
+          list.appendChild(innerListItem);
+        }
+      }
+    }
+    wrapper.appendChild(list);
+
+    this.shadowRoot.appendChild(styleElem);
+    this.shadowRoot.appendChild(wrapper);
+  }
+}
+
+customElements.define('directions-info', Directions);
+
+/*********************************************************************/
+/***                       Helper Functions:                       ***/
+/***          Shout out to the TA's lemme just yoink these         ***/
+/*********************************************************************/
+
+/**
+ * Recursively search for a key nested somewhere inside an object
+ * @param {Object} object the object with which you'd like to search
+ * @param {String} key the key that you are looking for in the object
+ * @returns {*} the value of the found key
+ */
+ function searchForKey(object, key) {
+    var value;
+    Object.keys(object).some(function (k) {
+      if (k === key) {
+        value = object[k];
+        return true;
+      }
+      if (object[k] && typeof object[k] === 'object') {
+        value = searchForKey(object[k], key);
+        return value !== undefined;
+      }
+    });
+    return value;
+  }
+  
+  /**
+   * Similar to getUrl(), this function extracts the organizations name from the
+   * schema JSON object. It's not in a standard location so this function helps.
+   *
+   * @param {Object} data Raw recipe JSON to find the org string of
+   * @returns {String} If found, it retuns the name of the org as a string, otherwise null
+   */
+   function getOrganization(data) {
+    if (data.publisher?.name) return data.publisher?.name;
+    if (data['@graph']) {
+      for (let i = 0; i < data['@graph'].length; i++) {
+        if (data['@graph'][i]['@type'] == 'Organization') {
+          return data['@graph'][i].name;
+        }
+      }
+    };
+    return null;
+  }
+  
+  /**
+   * Similar to getOrganization(), this extracts recipe name from raw JSON
+   * @param {Object} Data Raw recipe JSON to find name of
+   * @returns {String} if found, returns the name of recipe as string, otherwise null
+   */
+  function getRecipeTitle(data) {
+    if (data.name) return data.name;
+    let value = null;
+    if (data['@graph']) {
+      data['@graph'].forEach((obj) => {
+        if (obj['@type'] == 'Recipe') {
+          value = obj['name'];
+        }
+      });
+    }
+    return value;
+  }
+  
+  /**
+   * Extract the URL from the given recipe schema JSON object
```

Comment:
Codacy has a fix for the issue: [Sentence must end with a period.](https://app.codacy.com/gh/cse110-fa21-group27/cse110-fa21-group27/pullRequest?prid=8447291)
```suggestion
   * Extract the URL from the given recipe schema JSON object.
```